src/commands/Minion/farm.ts

```diff
@@ -0,0 +1,313 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+
+import {
+	stringMatches,
+	formatDuration,
+	rand,
+	itemNameFromID,
+	removeItemFromBank
+} from '../../lib/util';
+import { BotCommand } from '../../lib/BotCommand';
+import { Time, Activity, Tasks } from '../../lib/constants';
+import { FarmingActivityTaskOptions } from '../../lib/types/minions';
+import addSubTaskToActivityTask from '../../lib/util/addSubTaskToActivityTask';
+import Farming from '../../lib/skilling/skills/farming/farming';
+import { UserSettings } from '../../lib/settings/types/UserSettings';
+import { SkillsEnum } from '../../lib/skilling/types';
+import { minionNotBusy, requiresMinion } from '../../lib/minions/decorators';
+import resolvePatchTypeSetting from '../../lib/farming/functions/resolvePatchTypeSettings';
+import { PatchTypes } from '../../lib/farming';
+import bankHasItem from '../../lib/util/bankHasItem';
+import itemID from '../../lib/util/itemID';
+import { calcNumOfPatches } from '../../lib/skilling/functions/calcsFarming';
+import hasArrayOfItemsEquipped from '../../lib/gear/functions/hasArrayOfItemsEquipped';
+
+export default class extends BotCommand {
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			altProtection: true,
+			oneAtTime: true,
+			cooldown: 1,
+			usage: '[quantity:int{1}|name:...string] [plantName:...string]',
+			usageDelim: ' '
+		});
+	}
+
+	@minionNotBusy
+	@requiresMinion
+	async run(msg: KlasaMessage, [quantity, plantName = '']: [null | number | string, string]) {
+		if (msg.flagArgs.plants) {
+			return msg.channel.sendFile(
+				Buffer.from(
+					Farming.Plants.map(
+						plant =>
+							`${plant.seedType}: ${plant.name} -- lvl ${
+								plant.level
+							}: ${Object.entries(plant.inputItems)
+								.map(entry => `${entry[1]} ${itemNameFromID(parseInt(entry[0]))}`)
+								.join(', ')}\n	Default # of patches: ${
+								plant.defaultNumOfPatches
+							}\n${plant.additionalPatchesByFarmLvl.map(
+								entry =>
+									`	| Farming Level: ${entry[0]} => Total Additional Patches: ${entry[1]} |\n`
+							)} ${plant.additionalPatchesByQP.map(
+								entry =>
+									`	| Quest Points: ${entry[0]} => Total Additional Patches: ${entry[1]} |\n`
+							)} `
+					).join('\n')
+				),
+				`Farming Plants.txt`
+			);
+		}
+
+		await msg.author.settings.sync(true);
+		const userBank = msg.author.settings.get(UserSettings.Bank);
+		const questPoints = msg.author.settings.get(UserSettings.QP);
+		const GP = msg.author.settings.get(UserSettings.GP);
+		const currentWoodcuttingLevel = msg.author.skillLevel(SkillsEnum.Woodcutting);
+		const currentDate = new Date().getTime();
+
+		let payment = false;
+		let upgradeType = '';
+		let str = '';
```

Comment:
`str` is a bad name. You should use a descriptive name. Dont be afraid of making it long.